core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/PythonUDFOperator.java

```diff
@@ -0,0 +1,700 @@
+        for (Attribute a : tuple.getSchema().getAttributes()) {
+            String name = a.getName();
+            // When it is null, skip it.
+            if (tuple.getField(name).getValue() == null) continue;
+            switch (a.getType()) {
+                case INTEGER:
+                    ((IntVector) schemaRoot.getVector(name)).setSafe(index, (int) tuple.getField(name).getValue());
+                    break;
+                case DOUBLE:
+                    ((Float8Vector) schemaRoot.getVector(name)).setSafe(index, (double) tuple.getField(name).getValue());
+                    break;
+                case BOOLEAN:
+//                    ((BitVector) schemaRoot.getVector(name)).setSafe(index, ((Integer) tuple.getField(name).getValue()));
+//                    break;
+                case TEXT:
+                case STRING:
+                case _ID_TYPE:
+                    ((VarCharVector) schemaRoot.getVector(name)).setSafe(
+                            index, tuple.getField(name).getValue().toString().getBytes(StandardCharsets.UTF_8));
+                    break;
+                case DATE:
+                    ((DateDayVector) schemaRoot.getVector(name)).setSafe(index,
+                            (int)((LocalDate) tuple.getField(name).getValue()).toEpochDay());
+                    break;
+                case DATETIME:
+                    StructVector dateTimeStructs = ((StructVector) schemaRoot.getVector(name));
+                    if (tuple.getField(name).getValue() != null) {
+                        dateTimeStructs.setIndexDefined(index);
+                        DateDayVector subVectorDay = (DateDayVector) dateTimeStructs.getVectorById(0);
+                        TimeSecVector subVectorTime = (TimeSecVector) dateTimeStructs.getVectorById(1);
+                        LocalDateTime value = (LocalDateTime) tuple.getField(name).getValue();
+                        subVectorDay.setSafe(index, (int) value.toLocalDate().toEpochDay());
+                        subVectorTime.setSafe(index, value.toLocalTime().toSecondOfDay());
+                    }
+                    else dateTimeStructs.setNull(index);
+                    break;
+                case LIST:
+                    // For now only supporting span.
+                    if (((ImmutableList) tuple.getField(name).getValue()).get(0).getClass() != Span.class) {
+                        schemaRoot.clear();
+                        closeClientAndServer();
+                        throw (new DataflowException("Unsupported Element Type for List Field!"));
+                    }
+                    else {
+                        ListVector listVector = (ListVector) schemaRoot.getVector(name);
+                        ImmutableList<Span> spansList = (ImmutableList<Span>) tuple.getField(name).getValue();
+                        try {
+                            convertListOfSpans(spansList, listVector, index, name);
```

Comment:
seems we are still transmitting span list here. Are we going to change the span list to string in a following PR? Or is it already implemented in somewhere else that I missed?

Reply:
I've already changed this with a list of String in `convertListOfSpans` and `getSpanFromListVector`.